File: io/lucamienert/basic/ast/VariableExpression.java
package io.lucamienert.basic.ast;

import io.lucamienert.basic.instruction.Instruction;
import io.lucamienert.basic.instruction.InstructionSequence;
import io.lucamienert.basic.instruction.OpCode;

public class VariableExpression extends Expression {

    private final String name;

    public VariableExpression(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    @Override
    public void compile(InstructionSequence seq) {
        seq.append(new Instruction(OpCode.LOAD, name));
    }
    
}

File: src/BasicCompiler/Ast/VariableExpression.cs
using BasicCompiler.Instructions;

namespace BasicCompiler.Ast;

public class VariableExpression : Expression
{
    public string Name { get; set; }

    public VariableExpression(string name) 
        => Name = name;

    public override void Compile(InstructionSequence instructionSequence)
    {
        instructionSequence.Append(new Instruction(OpCode.LOAD, Name));
    }
}
